src/Workspaces/Core/Portable/CodeFixes/FixAllOccurrences/FixAllState.cs

```diff
@@ -43,65 +36,31 @@ internal sealed partial class FixAllState
             IEnumerable<string> diagnosticIds,
             FixAllContext.DiagnosticProvider fixAllDiagnosticProvider,
             CodeActionOptionsProvider codeActionOptionsProvider)
+            : base(fixAllProvider, document, project, codeFixProvider, scope, codeActionEquivalenceKey)
         {
-            Debug.Assert(document == null || document.Project == project);
-
             // We need the trigger diagnostic span for span based fix all scopes, i.e. FixAllScope.ContainingMember and FixAllScope.ContainingType
             Debug.Assert(diagnosticSpan.HasValue || scope is not FixAllScope.ContainingMember or FixAllScope.ContainingType);
 
-            FixAllProvider = fixAllProvider;
             DiagnosticSpan = diagnosticSpan;
-            Document = document;
-            Project = project;
-            CodeFixProvider = codeFixProvider;
-            Scope = scope;
-            CodeActionEquivalenceKey = codeActionEquivalenceKey;
             DiagnosticIds = ImmutableHashSet.CreateRange(diagnosticIds);
             DiagnosticProvider = fixAllDiagnosticProvider;
             CodeActionOptionsProvider = codeActionOptionsProvider;
         }
 
-        public Solution Solution => Project.Solution;
         internal bool IsFixMultiple => DiagnosticProvider is FixMultipleDiagnosticProvider;
 
-        public FixAllState WithScope(FixAllScope scope)
```

Comment:
Cleaned up all the With'ers and now have a single `With` method with optional parameters